### PR TITLE
fix: fix rule use function

### PR DESCRIPTION
**Pull request by LingRen**
[#8406](https://github.com/webpack/webpack/issues/8406)

### CODE REVIEW

**Comment by LingRen** [comment]
`rules: [
      {
        test: /\.css$/,
        // use: [{loader: 'style-loader'}, {loader: 'css-loader', options: {url: true}}],
        use: (info) => {
          console.log(info);

          return [
            {
              loader: 'style-loader'
            },
            {
              loader: 'css-loader',
              options: {
                url: true
              }
            }
          ]
        }
      }
    ]`

define function 
Module not found: Error: Can't find options with ident 'ref--xxx'

**Comment by evilebottnawi** [comment]
@LingRen need tests

**Comment by hiroppy** [comment]
friendly ping @LingRen

File: test/configCases/rule-set/test-user-fn/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+module.exports = {
+	module: {
+		rules: [
+			{
+				test: /\.css$/,
+				// use: [{loader: 'style-loader'}, {loader: 'css-loader', options: {url: true}}],
```
Comment (aaren-cordova) [inline test/configCases/rule-set/test-user-fn/webpack.config.js]: Did you intend to leave this comment here?

File: test/configCases/rule-set/test-user-fn/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+				use: info => {
+					console.log(info);
```
Comment (aaren-cordova) [inline test/configCases/rule-set/test-user-fn/webpack.config.js]: Did you mean to leave this `console.log` here?

Comment (LingRen) [reply test/configCases/rule-set/test-user-fn/webpack.config.js]: just to see print

**Comment by sokra** [comment]
done in `next` branch

**Comment by vvanpo** [comment]
@sokra Can you point me to the commit(s) where this was fixed, and is there a possibility of it being backported? As far as I can tell this bug still exists in v4.

**Pull request closed without merging**